src/query/storages/fuse/src/operations/common/mutation_log.rs

```diff
@@ -42,6 +43,8 @@ pub enum MutationLogEntry {
 pub struct ReplacementLogEntry {
     pub index: BlockMetaIndex,
     pub op: Replacement,
+    // for delete operation, delete whole segment
+    pub deleted_segment: Option<MutationDeletedSegment>,
```

Comment:
shall we convert `ReplacementLogEntry` into Enum? with two variants, one for block level replace, one for segment-level deletion?